src/app/components/users/users.component.tsx

```diff
@@ -27,74 +27,67 @@ const Users = (props: UserProps): JSX.Element => {
   )
 }
 
-class UsersComponent extends React.Component<UserComponentProps, UserState>{
-  constructor(props: UserComponentProps) {
-    super(props);
-    this.state = {
-      componentState: ComponentViewState.DEFAULT,
-    }
-    this.fetchUsers = this.fetchUsers.bind(this);
-    this.logout = this.logout.bind(this);
-  }
+const UsersComponent = (props: UserComponentProps): JSX.Element => {
+  const dependencies = React.useContext(DIContext);
+
+  const {translation, authService, userService} = dependencies;
+  const [state, setComponentState] = React.useState<UserState>({
+    componentState: ComponentViewState.DEFAULT
+  });
 
-  logout(): void {
-    this.props.authService.logout();
+  const {componentState, users, error} = state;
+  const isLoaded = componentState === ComponentViewState.LOADED;
+  const isLoading = componentState === ComponentViewState.LOADING;
+  const isError = componentState === ComponentViewState.ERROR;
+
+  const logout = (): void => {
+    authService.logout();
     // redirect to login page
-    this.props.history.push('/');
+    props.history.push('/');
   }
 
-  async fetchUsers(): Promise<void> {
-    this.setState({
-      componentState: ComponentViewState.LOADING,
-    });
-    const response = await this.props.userService.getUsers();
+  const fetchUsers = async(): Promise<void> => {
+    setComponentState({componentState: ComponentViewState.LOADING});
+    const response = await userService.getUsers();
     if (response.hasData()
       && response.data) {
-      this.setState({
+      setComponentState({
         componentState: ComponentViewState.LOADED,
         users: response.data,
       });
     } else {
-      //const msg = response.error || this.translate('no_internet');
-      this.setState({
+      const msg = response.error || translation.t('no_internet');
+      setComponentState({
         componentState: ComponentViewState.ERROR,
-        error: response.error,
+        error: msg,
       });
     }
   }
-  async componentDidMount(): Promise<void> {
-    this.fetchUsers();
-  }
-
-  render(): React.ReactNode {
-    const {componentState, users, error} = this.state;
-    const {translation} = this.props;
 
-    const isLoaded = componentState === ComponentViewState.LOADED;
-    const isLoading = componentState === ComponentViewState.LOADING;
-    const isError = componentState === ComponentViewState.ERROR;
+  useAsyncEffect(async (): Promise<void> => {
```

Comment:
How is loading, loaded, error works here?

Reply:
It's in this component, see `fetchUsers` function